### PR TITLE
feat: add proxy for invoice generation

### PR DESCRIPTION
**Pull request by im-adithya**
Adds /generate-invoice endpoint

takes `ln`, `amount`, `comment`

Screenshot 2023-03-10 at 10 16 17 PM

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -126,6 +130,55 @@ func main() {
...
+		callback := lnurlp.(map[string]interface{})["callback"]
+
+		// if the lnurlp response doesn't have a callback to generate invoice
+		if callback == nil {
+			return c.JSON(http.StatusBadRequest, &responseBody)
+		}
+
+    invoiceParams := c.QueryParam("invoiceParams")
+
+		invoice, invoiceResponse, err := GetJSON(callback.(string) + "?" + invoiceParams);
```
Comment (bumi) [inline main.go:165]: does this work? is `invoiceParams` an object? 
I would think of the URL as something like this: 
```
proxyurl.com?ln=satoshi@getalby.com&invoice[amount]=1000&invoice[comment]=foo%20bar
```
and then we pass on everything that's in `invoice`

is this how this works ?

Comment (bumi) [reply main.go:165]: otherwise I fear it is double encoded. and it is harder to first convert the invoice params into a string

Comment (im-adithya) [reply main.go:165]: Done!
Now it takes in `http://localhost:3000/generate-invoice?ln=adithya@getalby.com&amount=1000&comment=foo`

And deletes the `ln` param, encodes the rest again and appends it to the callback

File: main.go [hunk 2]
```diff
@@ -126,6 +130,56 @@ func main() {
     return c.JSONPretty(http.StatusOK, &responseBody, "  ")
   })
 
+  e.GET("/generate-invoice", func(c echo.Context) error {
```
Comment (rolznz) [inline main.go:133]: can this be a POST rather than a GET? GETs should be idempotent

Comment (bumi) [reply main.go:133]: LNURL uses GET :)

Comment (rolznz) [reply main.go:133]: Thanks, I will update my README PR

**Pull request merged**